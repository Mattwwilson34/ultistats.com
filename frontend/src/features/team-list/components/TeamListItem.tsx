import * as React from 'react'
import ListItem from '@mui/material/ListItem'
import ListItemButton from '@mui/material/ListItemButton'
import ListItemText from '@mui/material/ListItemText'

interface TeamListItemProps {
  teamName: string
}

export default function TeamListItem({
  teamName,
}: TeamListItemProps): React.ReactElement {
  return (
    <ListItem disablePadding>
      <ListItemButton>
        <ListItemText primary={teamName} />
      </ListItemButton>
    </ListItem>
  )
}
