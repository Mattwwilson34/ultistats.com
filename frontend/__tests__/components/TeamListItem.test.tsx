import React from 'react'
import { render, screen } from '@testing-library/react'
import TeamListItem from '../../src/features/team-list/components/TeamListItem'

describe('TeamListItem component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<TeamListItem teamName="Team A" />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders the team name', () => {
    const teamName = 'Team A'
    render(<TeamListItem teamName={teamName} />)
    const teamNameText = screen.getByText(teamName)
    expect(teamNameText).toBeInTheDocument()
  })

  it('does not render any additional text', () => {
    const teamName = 'Team A'
    render(<TeamListItem teamName={teamName} />)
    const additionalText = screen.queryByText(/Team [BCDEFGH]/)
    expect(additionalText).not.toBeInTheDocument()
  })
})
