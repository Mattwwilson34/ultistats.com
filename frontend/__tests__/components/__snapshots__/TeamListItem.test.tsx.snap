// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamListItem component matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
  >
    <div
      class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          Team A
        </span>
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
  </li>
</DocumentFragment>
`;
